src/java/org/pantsbuild/tools/junit/impl/ConsoleRunnerImpl.java

```diff
@@ -570,24 +576,19 @@ private int runLegacy(Collection<Spec> parsedTests, JUnitCore core) throws Initi
     if (!classes.isEmpty()) {
       if (this.perTestTimer || this.parallelThreads > 1) {
         for (Class<?> clazz : classes) {
-          if (legacyShouldRunParallelMethods(clazz)) {
-            if (ScalaTestUtil.isScalaTestTest(clazz)) {
-              // legacy and scala doesn't work easily.  just adding the class
-              requests.add(new AnnotatedClassRequest(clazz, numRetries, err));
-            } else {
-              testMethods.addAll(TestMethod.fromClass(clazz));
-            }
+          // legacy doesn't support scala test tests, so those are run as classes.
+          if (legacyShouldRunParallelMethods(clazz) && !ScalaTestUtil.isScalaTestTest(clazz)) {
+            testMethods.addAll(TestMethod.fromClass(clazz));
           } else {
-            requests.add(new AnnotatedClassRequest(clazz, numRetries, err));
+            requests.add(createAnnotatedClassRequest(err, clazz));
```

Comment:
Also a change of behaviour; before a request wasn't added if `!legacyShouldRunParallelMethods(clazz)` but now it is? Intentional?

Reply:
I think it shouldn't be a change of behavior. I put together a truth table when I was trying to be sure before I changed it. It looks something like this

bad psuedojavacode follows
```
L = legacyShouldRunParallelMethods(clazz)
S = ScalaTestUtil.isScalaTestTest(clazz)
radd =  requests.add( annotated request)
madd = testMethods.addAll(TestMethod.fromClass(clazz))
```
original logic
```
if (L) {
  if(S){
    radd
  } else {
    madd
  }
} else {
  radd
}
```
I built a truth table like this, and found that it looked a bit like L && !S, so I reduced the nested ifs.

```
| L | S | L && !S | result  |
-----------------------------
| T | T |  F      | radd    |
| T | F |  T      | madd    |
| F | T |  F      | radd    |
| F | F |  F      | radd    |
```

```
if (L&&!S) { testmeth } else { reqadd }
```

Reply:
I was apparently super bad at reading when I reviewed this :( Sorry again!